src/Microsoft.Framework.DesignTimeHost/DesignTimeAssemblyLoadContextFactory.cs

```diff
@@ -31,23 +31,18 @@ internal class DesignTimeAssemblyLoadContextFactory : IAssemblyLoadContextFactor
 
         public IAssemblyLoadContext Create(IServiceProvider serviceProvider)
         {
-            return GetRuntimeFactory(serviceProvider).Create(serviceProvider);
-        }
-
-        private IAssemblyLoadContextFactory GetRuntimeFactory(IServiceProvider serviceProvider)
-        {
-            var cacheKey = Tuple.Create("RuntimeLoadContextFactory", _project.Name, _appEnv.Configuration, _appEnv.RuntimeFramework);
-
-            return _cache.Get<IAssemblyLoadContextFactory>(cacheKey, ctx =>
+            var hostContextKey = Tuple.Create("RuntimeLoadContext", _project.Name, _appEnv.Configuration, _appEnv.RuntimeFramework);
+            
+            var appHostContext = _cache.Get<ApplicationHostContext>(hostContextKey, ctx => 
```

Comment:
Oh dur, of course since there's a lambda. Kinda odd API name though `Get` instead of `GetOrAdd`.